sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -12,23 +12,62 @@ import { ProxySettings } from "../serviceClient";
 import { WebResourceLike } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
+import { getEnvironmentValue } from "../util/utils";
+
+let noProxyList: string[] = [];
+let isNoProxyInitalized = false;
+let byPassedList = new Map();
 
 function loadEnvironmentProxyValue(): string | undefined {
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  const httpsProxy =
+    getEnvironmentValue(Constants.HTTPS_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+  const httpProxy =
+    getEnvironmentValue(Constants.HTTP_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+
+  return httpsProxy || httpProxy;
+}
+
+// Check whether uri match noProxyList. If match, won't set proxy settings while sending request to the uri.
```

Comment:
Is this better? Feel free to reject this suggestion.

```suggestion
// Check whether the given `uri` matches the noProxyList. If it matches, any request sent to that same `uri` won't set the proxy settings.
```